client/client.go

```diff
@@ -139,7 +140,11 @@ func (c *CSAPI) CreateRoom(t ct.TestLike, body map[string]interface{}) *http.Res
 }
 
 // MustJoinRoom joins the room ID or alias given, else fails the test. Returns the room ID.
-func (c *CSAPI) MustJoinRoom(t ct.TestLike, roomIDOrAlias string, serverNames []string) string {
+//
+// Args:
+//   - `serverNames`: The list of servers to attempt to join the room through.
+//     These should be a resolvable addresses within the deployment network.
+func (c *CSAPI) MustJoinRoom(t ct.TestLike, roomIDOrAlias string, serverNames []spec.ServerName) string {
```

Comment:
Basically, wherever we expect people to use `deployment.GetFullyQualifiedHomeserverName(t, hsName)`, I've updated these function signatures to accept `spec.ServerName` instead of just plain strings.

I think this is also more semantically correct for the places that this needs to be a resolvable address in the federation.